services/pypi/pypi-pyversions.service.js

```diff
@@ -58,6 +58,15 @@ module.exports = class PypiPythonVersions extends PypiBase {
       packageData,
       /^Programming Language :: Python :: ([\d.]+)$/
     )
+    // If no versions are found yet, check "X :: Only" as a fallback.
+    if (versions.length === 0) {
+      versions.push(
+        ...parseClassifiers(
+          packageData,
+          /^Programming Language :: Python :: (\d+) :: Only$/
```

Comment:
Is something gained by this conditional? Would it be simpler to remove the conditional, and optionally match `:: Only` in a single pass?

Reply:
The conditional is in my opinion desirable. For example, given `3 :: Only`, `3.5`, `3.6`, and `3.7`, I think the "correct" output would be `3.5 | 3.6 | 3.7`.  But for `3`, `3.5`, `3.6`, and `3.7`, it would be `3 | 3.5 | 3.6 | 3.7`. So the `X :: Only` version would not be taken to include all `X.*` versions if more specific version info is available.

Reply:
That would be a good example to add a test for. It really clarifies the intent of the code.

Reply:
Yep, the `"Only" and others` test case for uvloop 0.12.1 was already included in PR for this reason, it mostly covers it.